zk/groth16/src/proof/mod.rs

```diff
@@ -30,6 +31,18 @@ impl<E: Pairing> From<&Proof<E>> for ark_groth16::Proof<E> {
         }
     }
 }
+
+impl<E: Pairing> From<ark_groth16::Proof<E>> for Proof<E> {
```

Comment:
Why this? Will we leak the ark one or planning to use it? Is it related to de/serialization?

Reply:
Yes this is to serialize a proof (also we already have the `From` in the other direction so we're not leaking more stuff). We could also derive serialization / deserialization internally if you prefer

Reply:
Check on the lately added Compressed form of the proof, which is what we need to use for serialization/deserialization:
https://github.com/logos-co/nomos/pull/1674

Reply:
Serde is not derived for it tho. I would prefer to have an external type for serializing it where needed. Although I understand it may be easier to derive here. If we chose to derive, lets feature gate it (?)